library/core/src/time.rs

```diff
@@ -208,6 +222,9 @@ impl Duration {
     #[inline]
     #[must_use]
     #[rustc_const_stable(feature = "duration_consts_2", since = "1.58.0")]
+    // by definition of NANOS_PER_SEC, the checks for div by 0 cases are unreachable, but also unneeded
```

Comment:
I admit that I mostly find this comment confusing. It's a comment about a detail at a point where, as a reader, I am trying to get the bigger picture first. Does this comment really add value? Perhaps you should, instead, just spell this out as a precondition?